NMS-12391: Use bulkhead to restrict number of threads (#3005)
OpenNMS/opennms: opennms-doc/guide-admin/src/asciidoc/text/minion/kafka-rpc.adoc

// Allow GitHub image rendering
:imagesdir: ../../images

===  Tuning Kafka RPC on Minion

To avoid too many requests hogging the system at once, Kafka RPC on Minion limits the maximum number of concurrent RPC requests.

> Currently max number of concurrent requests is set to 1000 with maximum wait time of 100ms.

Reviewer: ```suggestion
Currently the maximum number of concurrent requests is set to 1,000 with a maximum wait time of 100ms.
```

> With these settings when concurrent requests reach 1000, it only allow 10 extra requests per sec.

Reviewer: ```suggestion
With these settings, when concurrent requests reach 1000, Kafka RPC will allow only 10 extra requests per second.
```

> These settings can be tuned as below

Reviewer: ```suggestion
Tune these settings as below:
```
